nixos/modules/security/doas.nix

```diff
@@ -236,57 +245,67 @@ in
 
   ###### implementation
 
-  config = mkIf cfg.enable {
-
-    security.doas.extraRules = mkOrder 600 [
-      {
-        groups = [ "wheel" ];
-        noPass = !cfg.wheelNeedsPassword;
-      }
-    ];
-
-    security.wrappers.doas =
-      { setuid = true;
-        owner = "root";
-        group = "root";
-        source = "${doas}/bin/doas";
-      };
-
-    environment.systemPackages = [
-      doas
-    ];
-
-    security.pam.services.doas = {
-      allowNullPassword = true;
-      sshAgentAuth = true;
-    };
-
-    environment.etc."doas.conf" = {
-      source = pkgs.runCommand "doas-conf"
+  config = mkMerge [
```

Comment:
We cannot mkMerge config as it easily creates infinite recursions. Please write the attr out twice or nest the conditions.